### ISSUE TITLE
PSR-12 is in ithe making of creating a style guide for attributes

### ISSUE THREAD

**Issue by snake-py** [squizlabs/PHP_CodeSniffer#3688]
Hello,

today I noticed that code sniffer (with psr-12) will throw an error by code like this:

```php
#[AsCommand(name: 'serveDebug')]
class ServeDebug extends ServeCommand
```

because the current rule seems not to take into account that one line above a class definition can be attributed the output: 

```php

FILE: /app/Console/Commands/ServeDebug.php
----------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 6 | ERROR | [x] Header blocks must be separated by a single blank
   |       |     line
----------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------
```

I know that they have not totally settled on a style https://github.com/php-fig/per-coding-style/pull/26 

I just wanted to raise awareness, because I am guessing the rule should be changed for php 8 users. I don't see that they will settle on an required additional line between the attribute and the class.

**Comment by jrfnl** [comment]
Related to #3793

**Comment by rodrigoprimo** [comment]
I was not able to reproduce this problem on the latest version of PHPCS using the PSR12 standard. Here is the output that I got using a file with the code sample provided here:

```
❯ bin/phpcs --standard=PSR12 test.php                                                                                                                                                                                                                                                                            at 17:00:49 
Time: 25ms; Memory: 6MB
```

It seems to me this is a duplicate of #3286, which was closed as fixed. I believe that this issue can be closed.
